### PR TITLE
fix(total-balance): Reuse TotalBalance from Home screen in FiatExchange Screen

### PR DESCRIPTION
**Pull request by jh2oman**
### Description

[New Designs
](https://www.figma.com/file/8USELhisKuP1YlCt95WrD2/Support-for-Multiple-Tokens?node-id=1802%3A60858)

The FiatExchange screen was using a very simplistic and outdated calculation of total value which included just Cusd. It now reflects the value of the homescreen with a slightly different style

## New
Screen Shot 2022-02-08 at 3 13 33 PM

## Old
Screen Shot 2022-02-08 at 3 22 06 PM

### Other changes

Renamed HomeTotalBalance file to TotalBalance to make it more versatile. 

### Tested

Visually and with new tests.

### How others should test

Verify that the Home screen and Withdraw screen show the same Total Value

### Related issues

- Fixes [[Bug] Add funds current balance in the withdraw flow includes cUSD only](https://app.zenhub.com/workspaces/acquisition-squad-sprint-board-6010683afabec1001a090887/issues/valora-inc/wallet/1908)

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

File: packages/mobile/src/components/TokenBalance.test.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    exchangeRates: {
+      [Currency.Dollar]: '1',
+      [Currency.Euro]: null,
+      [Currency.Celo]: null,
+    },
+  },
+}
+
+describe('FiatExchangeTokenBalance and HomeTokenBalance', () => {
+  it.each([HomeTokenBalance, FiatExchangeTokenBalance])(
```
Comment (jh2oman) [inline packages/mobile/src/components/TokenBalance.test.tsx:26]: This looks like I added a lot of new tests, but the only thing I did was add the `it.each` to every test and remove the `expect(tree).toMatchSnapshot()` checks

**Review by gnardini — APPROVED** [review]
Minor nit, I do think we should probably remove the `useErrorMessageWithRefresh` call from the CICO screen but the rest looks great, approving :)

File: packages/mobile/src/components/TokenBalance.tsx [hunk 2]
```diff
@@ -145,6 +145,39 @@ function HomeTokenBalance() {
   )
 }
 
+export function FiatExchangeTokenBalance() {
```
[the remark this answers is not in the document]
Comment (jh2oman) [reply packages/mobile/src/components/TokenBalance.tsx:148]: Its not my favorite either, but their functionality and style was so similar that I thought the grouping was acceptable.

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified the above issue in Latest Android Internal Release build V:1.26.0 and updated the status in test rails. 

**Test rail link:**
**Android 8:** https://valoraapp.testrail.io/index.php?/tests/view/9820&group_by=cases:section_id&group_order=asc&group_id=35

**Android 10:** https://valoraapp.testrail.io/index.php?/tests/view/9946&group_by=cases:section_id&group_order=asc&group_id=35

**Device:** Samsung Galaxy A8+(8.0), Samsung Galaxy M40(10.0)

**Cc:** @jh2oman 
Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified the above issue on Latest Test flight Release Build V1.26.0 and updated status in Test rails.

**Test rail link:**
**iOS 13:** https://valoraapp.testrail.io/index.php?/tests/view/10012&group_by=cases:section_id&group_order=asc&group_id=35

**Device:** iPhone 7(13.3)

**CC:** @jh2oman 
Thanks..!!

**Pull request merged**